pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -1040,3 +1048,36 @@ func checkForFailureConditions(cr *minterv1.CredentialsRequest) bool {
 	}
 	return false
 }
+
+// UpdateProvisionedStatus will update the status subresource of this CredentialsRequest
+func (r *ReconcileCredentialsRequest) UpdateProvisionedStatus(cr *minterv1.CredentialsRequest, provisioned bool) error {
+	// Check if the status subresource is already set
+	if cr.Status.LastSyncTimestamp == nil {
+		// Create a new status object and set its fields
+		cr.Status = minterv1.CredentialsRequestStatus{
+			LastSyncTimestamp: &metav1.Time{Time: time.Now()},
+			Provisioned:       provisioned,
+			ProviderStatus:    &runtime.RawExtension{},
+		}
+	} else {
+		// Update the Provisioned field with the given parameter
+		cr.Status.Provisioned = provisioned
+	}
+
+	// Use UpdateProvisionedStatus method to update only the status subresource
+	err := r.Client.Status().Update(context.Background(), cr)
+	if err != nil {
+		// Handle possible errors from the API server
+		if errors.IsConflict(err) {
+			// Conflict error means that the object has been modified by another process
+			// Get the latest version of the object and retry the update operation
+			err := r.Client.Get(context.Background(), types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, cr)
+			if err != nil {
+				return err
+			}
+			return r.UpdateProvisionedStatus(cr, provisioned)
+		}
+		return err
+	}
```

Comment:
I'm wondering if we could use retry.RetryOnConflict() here

Reply:
Done in [4f25b93](https://github.com/openshift/cloud-credential-operator/pull/562/commits/4f25b9368c7c03d1ea90f469ced8162dd70e6653) Thanks Feilian, yeah, no sense reinventing the wheel!